backend/dataall/modules/shares_base/db/share_object_state_machines.py

```diff
@@ -0,0 +1,296 @@
+import logging
+
+from dataall.base.db import exceptions
+from dataall.modules.shares_base.services.shares_enums import (
+    ShareObjectActions,
+    ShareObjectStatus,
+    ShareItemActions,
+    ShareItemStatus,
+)
+from dataall.modules.shares_base.db.share_object_repositories import ShareObjectRepository
+
+logger = logging.getLogger(__name__)
+
+
+class Transition:
+    def __init__(self, name, transitions):
+        self._name = name
+        self._transitions = transitions
+        self._all_source_states = [*set([item for sublist in transitions.values() for item in sublist])]
+        self._all_target_states = [item for item in transitions.keys()]
```

Comment:
nit: why is this needed? what's wrong with `self._all_target_states = transitions.keys()`?